dora/core/server/worker/src/main/java/alluxio/worker/netty/FileReadHandler.java

```diff
@@ -90,63 +77,93 @@ public DoraWorker getWorker() {
   }
 
   /**
-   * The packet reader to read from a local block worker.
+   * Factory for creating {@link FilePacketReader}s.
    */
-  @NotThreadSafe
-  public class BlockPacketReader extends PacketReader {
-    /**
-     * The Dora Worker.
-     */
+  public static final class FilePacketReaderFactory
+      implements PacketReader.Factory<BlockReadRequest, FilePacketReader> {
     private final DoraWorker mWorker;
+    private final FileTransferType mTransferType;
 
     /**
-     * Creates an instance of {@link BlockPacketReader}.
+     * Constructor.
      *
-     * @param worker dora worker
+     * @param doraWorker dora worker
+     * @param transferType transfer type
      */
-    public BlockPacketReader(DoraWorker worker) {
-      mWorker = worker;
+    public FilePacketReaderFactory(DoraWorker doraWorker, FileTransferType transferType) {
+      mWorker = doraWorker;
+      mTransferType = transferType;
     }
 
     @Override
-    public void completeRequest(BlockReadRequestContext context) throws Exception {
-      BlockReader reader = context.getBlockReader();
-      if (reader != null) {
-        try {
-          reader.close();
-        } catch (Exception e) {
-          LOG.warn("Failed to close block reader for block {} with error {}.",
-              context.getRequest(), e.getMessage());
+    public FilePacketReader create(BlockReadRequest readRequest) throws IOException {
+      try {
+        final String fileId =
+            new AlluxioURI(readRequest.getOpenUfsBlockOptions().getUfsPath()).hash();
+        BlockReader reader =
+            mWorker.createFileReader(fileId, readRequest.getStart(),
+                /* positionShort */ false, readRequest.getOpenUfsBlockOptions());
```

Comment:
The same thing that is confusing here, again.
We are creating a File Reader, and the result is a Block Reader.

The same question as previous one.